src/main/java/scalismo/image/BSplineCoefficients.java

```diff
@@ -76,7 +76,9 @@ static float getInitialCausalCoefficientMirrorOnBounds(float[] c, float z,
 	} /* end getInitialCausalCoefficientMirrorOnBounds */
 
 
-	
+    // TODO: not actually deprecated, but should be renamed to something that makes it clear that
+    // TODO: the method modifies data in place.
+    @Deprecated
 	static public void getSplineInterpolationCoefficients(int degree, float[] c) throws Exception {
```

Comment:
I think this is not really an issue. It is anyway obvious by the type signature (returns void). 
I would argue for just removing the deprecated and make it private (it was never intended for public use).